internal/envtest/multi_envtest.go

```diff
@@ -0,0 +1,26 @@
+package envtest
+
+import (
+	"testing"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+type MultiK8sEnvironment []*Environment
+
+func RunMulti(m *testing.M, setupFunc func(e *MultiK8sEnvironment), numClusters int) (code int) {
+	e := make(MultiK8sEnvironment, numClusters)
+	ctx := ctrl.SetupSignalHandler()
+	for i := 0; i < numClusters; i++ {
+		e[i] = NewEnvironment(ctx)
+		e[i].Start()
+	}
+	defer func() {
+		for i := 0; i < numClusters; i++ {
+			e[i].Stop()
+		}
+	}()
+	setupFunc(&e)
+	exitCode := m.Run()
```

Comment:
nitpick: Unnecessary exitCode, could just do ``return m.Run()``

Reply:
Then you'd have to defer the `Stop()`... This seems cleaner?

Reply:
No, it wouldn't change anything here. This exitCode is never used. I'm surprised our linting isn't catching this.